tests/phpunit/includes/testcase-xml.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+abstract class WP_Test_XML_TestCase extends WP_UnitTestCase {
+	/**
+	 * Load XML from a string.
+	 *
+	 * @param string $xml
+	 * @param int    $options Bitwise OR of the {@link https://www.php.net/manual/en/libxml.constants.php libxml option constants}.
+	 *                        Default is 0.
+	 * @return DOMDocument The DOMDocument object loaded from the XML.
+	 */
+	public function loadXML( $xml, $options = 0 ) {
+		// Suppress PHP warnings generated by DOMDocument::loadXML(), which would cause
+		// PHPUnit to incorrectly report an error instead of a just a failure.
+		$internal = libxml_use_internal_errors( true );
+		libxml_clear_errors();
+
+		$xml_dom = new DOMDocument();
+		$xml_dom->loadXML( $xml, $options );
+		$libxml_last_error = libxml_get_last_error();
+
+		$this->assertFalse(
+			isset( $libxml_last_error->message ),
+			isset( $libxml_last_error->message ) ? sprintf( 'Non-well-formed XML: %s.', $libxml_last_error->message ) : ''
+		);
+
+		// Restore default error handler.
+		libxml_use_internal_errors( $internal );
+		libxml_clear_errors();
+
+		return $xml_dom;
+	}
+
+	/**
+	 * Normalize an XML document to make comparing two documents easier.
+	 *
+	 * @param string $xml
+	 * @param int    $options Bitwise OR of the {@link https://www.php.net/manual/en/libxml.constants.php libxml option constants}.
+	 *                        Default is 0.
+	 * @return string The normalized form of `$xml`.
+	 */
+	public function normalizeXML( $xml, $options = 0 ) {
+		if ( ! class_exists( 'XSLTProcessor' ) ) {
```

Comment:
Created https://github.com/WordPress/wpdev-docker-images/pull/31.

Not sure if that is correct though.